### PR TITLE
feat(serviceaccount):- get owncompany serviceaccounts

### PR DESCRIPTION
**Pull request by qxz2mqe**
## Description

Added relation for serviceaccount id in response body 

## Why

Added relation to know whether service account belongs to owner or provider

## Issue

CPLP-2786

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [ ] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [ ] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs [hunk 1]
```diff
@@ -161,6 +161,7 @@ public ServiceAccountRepository(PortalDbContext portalDbContext)
                 serviceAccount.ClientClientId,
                 serviceAccount.Name,
                 serviceAccount.CompanyServiceAccountTypeId,
+                serviceAccount.CompaniesLinkedServiceAccount!.Provider != null ? "provider" : "owner",
```
Comment (Phil91) [inline src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs:164]: this needs either to be the value of provider or owner not a string value but instead the uuid of the database column of the view

File: tests/portalbackend/PortalBackend.DBAccess.Tests/ServiceAccountRespotitoryTests.cs [hunk 2]
```diff
@@ -202,6 +202,7 @@ public async Task GetOwnCompanyServiceAccountsUntracked_ReturnsExpectedResult(in
         if (expected > 0)
         {
             result.Data.First().CompanyServiceAccountTypeId.Should().Be(CompanyServiceAccountTypeId.OWN);
+            result.Data.First().Relation.Should().Be("owner");
```
Comment (Phil91) [inline tests/portalbackend/PortalBackend.DBAccess.Tests/ServiceAccountRespotitoryTests.cs:205]: Please adjust the test accordingly after changing the implementation

**Review by Phil91 — CHANGES_REQUESTED** [review]

**Comment by Phil91** [comment]
@ntruchsess could you please also have a look at it as well?
I'm not quite sure about the "migration" since we want to use the view to get the data of the provider / owner it seems to be needed to have a key (at least for ef to be able to create the navigation and links) i tryed different solutions by myself but didn't came up with something better.

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.Migrations.Migrations
+{
+    public partial class CPLP2786AddCompanyAccountServiceToView : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.cs:29]: It's sufficient to have the changes in DbContext, there is no point in creating an empty migration. Please remove.

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.Designer.cs [hunk 4]
```diff
@@ -0,0 +1,7080 @@
...
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+using Npgsql.EntityFrameworkCore.PostgreSQL.Metadata;
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities;
+
+#nullable disable
+
+namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.Migrations.Migrations
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.Designer.cs:32]: It's sufficient to have the changes in DbContext, there is no point in creating an empty migration. Please remove.

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/********************************************************************************
+﻿/********************************************************************************
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs:1]: It's sufficient to have the changes in DbContext, there is no point in creating an empty migration. Please revert those changes of the snapshot.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs (hunk 5 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs:1]: reverted

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.Designer.cs (hunk 4 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.Designer.cs]: reverted

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.cs (hunk 3 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20230713071711_CPLP-2786-AddCompanyAccountServiceToView.cs]: reverted

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs:164]: @Phil91  I have clarified with julia and it is string

File: tests/portalbackend/PortalBackend.DBAccess.Tests/ServiceAccountRespotitoryTests.cs (hunk 2 above)
Comment (qxz2mqe) [reply tests/portalbackend/PortalBackend.DBAccess.Tests/ServiceAccountRespotitoryTests.cs:205]: No change is required as confirm with julia

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs (hunk 1 above)
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs:164]: please create an enum-type 'CompaniesLinkedServiceAccountType' in DbAccess/Models and use that (Then database will return an int while the json-serialization will take care of converting the enum-value to string).

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs (hunk 5 above)
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs:1]: to revert checkout the previous version of the file. This one still has changes (it has changed to encoding 'UTF-8 with BOM' while it should be plain UTF-8)

File: src/portalbackend/PortalBackend.PortalEntities/Entities/CompanyServiceAccount.cs [hunk 6]
```diff
@@ -59,4 +60,5 @@ public CompanyServiceAccount(Guid id, string name, string description, CompanySe
     public virtual OfferSubscription? OfferSubscription { get; set; }
     public virtual Connector? Connector { get; set; }
     public virtual ICollection<AppInstanceAssignedCompanyServiceAccount> AppInstances { get; private set; }
+    public virtual CompaniesLinkedServiceAccount? CompaniesLinkedServiceAccount { get; set; }
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Entities/CompanyServiceAccount.cs:63]: set must not be public as - while the pk of this relationship is on the other side - this other side is a view that cannot be updated.

File: src/portalbackend/PortalBackend.PortalEntities/Views/CompaniesLinkedServiceAccount.cs [hunk 7]
```diff
@@ -18,11 +18,14 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Entities;
+
 namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Views;
 
 public class CompaniesLinkedServiceAccount
 {
     public Guid ServiceAccountId { get; set; }
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Views/CompaniesLinkedServiceAccount.cs:27]: all setters must be private as the view cannot be updated.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs (hunk 1 above)
Comment (Phil91) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs:164]: @ntruchsess had a discussion with julia about the string value, we agreed on changing it to a bool flag to decide whether it's a the provider or owner

Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs]: fixed

File: src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs (hunk 5 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.Migrations/Migrations/PortalDbContextModelSnapshot.cs]: fixed

File: src/portalbackend/PortalBackend.PortalEntities/Entities/CompanyServiceAccount.cs (hunk 6 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/CompanyServiceAccount.cs]: fixed

File: src/portalbackend/PortalBackend.PortalEntities/Views/CompaniesLinkedServiceAccount.cs (hunk 7 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.PortalEntities/Views/CompaniesLinkedServiceAccount.cs]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Models/CompanyServiceAccountData.cs [hunk 8]
```diff
@@ -28,7 +28,9 @@ public record CompanyServiceAccountData(
     [property: JsonPropertyName("clientId")] string? ClientId,
     [property: JsonPropertyName("name")] string Name,
     [property: JsonPropertyName("serviceAccountType")] CompanyServiceAccountTypeId CompanyServiceAccountTypeId,
+    [property: JsonPropertyName("relation")] string Relation,
```
[the remark this answers is not in the document]
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.DBAccess/Models/CompanyServiceAccountData.cs]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs [hunk 9]
```diff
@@ -161,7 +161,7 @@ public ServiceAccountRepository(PortalDbContext portalDbContext)
                 serviceAccount.ClientClientId,
                 serviceAccount.Name,
                 serviceAccount.CompanyServiceAccountTypeId,
-                serviceAccount.CompaniesLinkedServiceAccount!.Provider != null ? "provider" : "owner",
+                serviceAccount.CompaniesLinkedServiceAccount!.Provider != null ? false : true,
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/ServiceAccountRepository.cs]: did fix my own comment.

**Pull request merged**